toolkit/scss/forms/_selection-buttons.scss

```diff
@@ -36,39 +38,122 @@
+    @if ($is-ie == false) or ($ie-version == 9) {
+      .js-enabled & {
+        margin: 0;
+        @include opacity(0);
+      }
+    }
   }
-}
 
-.selection-button-boolean,
-.selection-button-inline {
-  clear: none;
-  margin-right: $gutter-half;
-}
+  .js-enabled & {
+    &.selection-button-radio::before {
+      content: "";
+      border: 2px solid;
+      background: transparent;
+      width: 34px;
+      height: 34px;
+      position: absolute;
+      top: 0;
+      left: 0;
+      @include border-radius(50%);
+    }
 
-// Selected and focused states
+    &.selection-button-radio::after {
+      content: "";
+      border: 10px solid;
+      width: 0;
+      height: 0;
+      position: absolute;
+      top: 9px;
+      left: 9px;
+      @include border-radius(50%);
+      @include opacity(0);
+    }
 
-// Add selected state
-.js-enabled label.selection-button-selected {
-  background: $white;
-  border-color: $black;
-}
+    &.selection-button-checkbox::before {
+      content: "";
+      border: 2px solid;
+      background: transparent;
+      width: 34px;
+      height: 34px;
+      position: absolute;
+      top: 0;
+      left: 0;
+    }
+
+    &.selection-button-checkbox::after {
+      content: "";
+      border: solid;
+      border-width: 0 0 5px 5px;
+      background: transparent;
+      width: 17px;
+      height: 7px;
+      position: absolute;
+      top: 10px;
+      left: 8px;
+      -moz-transform: rotate(-45deg); // Firefox 15 compatibility
+      -o-transform: rotate(-45deg); // Opera 12.0 compatibility
+      -webkit-transform: rotate(-45deg); // Safari 8 compatibility
+      -ms-transform: rotate(-45deg); // IE9 compatibility
+      transform: rotate(-45deg);
+      @include opacity(0);
+    }
+
+    // Focused state
+    &.selection-button-radio,
+    &.selection-button-checkbox {
+      &.focused::before {
+        @include box-shadow(0 0 0 5px $focus-colour);
+      }
+      // IE8 focus outline should stay as a border around the entire label
+      // Lack of padding doesn’t matter as IE8 won’t resize the inputs.
+      @include ie-lte(8) {
+        &.focused {
+          outline: 3px solid $focus-colour;
+
+          input:focus {
+            outline: none;
+          }
+        }
+      }
+    }
 
-// Add focus to block labels
-.js-enabled label.selection-button-focused {
-  outline: 3px solid $yellow;
+    // Selected state
+    &.selection-button-radio,
+    &.selection-button-checkbox {
+      &.selected::after {
+        @include opacity(1);
+      }
+    }
+  }
+
+  &:last-child,
+  &:last-of-type {
+    margin-bottom: 0;
+  }
+
+  .inline & {
+    clear: none;
+    margin-right: $gutter-half;
+  }
 }
 
-// Remove focus from radio/checkboxes
-.js-enabled .selection-button-focused input:focus {
-  outline: none;
+// To stack horizontally, use .inline on parent, to sit block labels next to each other
+.inline .block-label {
```

Comment:
Yup, makes sense.
Lose terseness but gain clarity. 

Read [a blog post by Harry Roberts](https://csswizardry.com/2017/02/code-smells-in-css-revisited/) the other day and that seems to be one of his ideas.